src/MicrosoftTeams.ts

```diff
@@ -417,25 +441,132 @@ namespace microsoftTeams
         {
             ensureInitialized(frameContexts.content, frameContexts.settings, frameContexts.remove);
 
+            // Open an authentication window with the parameters provided by the caller
+            openAuthenticationWindow(authenticateParameters.url, authenticateParameters.width, authenticateParameters.height);
+
+            authParams = authenticateParameters;
+        }
+
+        function closeAuthenticationWindow(): void
+        {
+            // Stop monitoring the authentication window
+            stopAuthenticationWindowMonitor();
+
+            // Try to close the authentication window and clear all properties associated with it
+            try
+            {
```

Comment:
I've seen this call throw before but I can't seem to reproduce it now. I mostly added the try...finally because I wanted to make sure that we always clean up the childWindow and childOrigin regardless of whether closing the window was successful or not. At this point I would prefer to let the exception bubble out so that I can make sure this call isn't failing in some unexpected cases. I might just completely remove the try...finally at some point.